app/src/main/java/com/github/kulebin/myfoursquareapp/dataSource/IDataSource.java

```diff
@@ -0,0 +1,16 @@
+package com.github.kulebin.myfoursquareapp.dataSource;
+
+public interface IDataSource {
+
+    //void fetchVenueList(IOnResultCallback pOnResultCallback);
+
+    <Result> void fetchData(String pUrl, IOnResultCallback<Result> pOnResultCallback);
```

Comment:
Your presenters and usecases shoundn't mind about the url.
Let this class has methods: fetchVenueList, fetchVenueDetail, etc. and provide necessary params